adapters/handlers/rest/raft/handler.go

```diff
@@ -0,0 +1,137 @@
+package raft
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gorilla/mux"
```

Comment:
Which features of gorilla are we using that make it better than net/http?  It look like our needs are very simple here, can we just use the built-in http server?

Reply:
good point, why can't we use golang's built-in http library?

Reply:
So I did a bit of research. Default golang mux doesn't look very efficient/promising. According to this benchmark https://github.com/julienschmidt/go-http-routing-benchmark#conclusions


They recommend their own router (it's their own benchmark so take it with a grain of salt) for performance (https://github.com/julienschmidt/httprouter).

However as they point out, their router is very barebones and does not include more features like mux (or others).

Golang default `ServeMux` also doesn't support method-based routing (POST/GET/etc..) or variables in URL paths. Will these be deal breaker in the future ?

From my own experience i'd be OK staying with gorilla/mux as it has feature we might wish to use in the future and their performance are OK. Open to talking about this :)

Reply:
There's a few points here:
* I don't trust any benchmark from someone who recommends their own product instead
* I don't see any indication that we are limited by the http router
* We don't do method based routing and in the event that we want to, adding `switch.Method() {` to our code is not hard
* I've had bad experiences with every router replacement and framework in go.
* Is this handling requests for GRPC?  Because we are using GRPC for speed, the benchmarks to beat aren't HTTP, they are the GRPC ones.

Reply:
I agree with your points. I think it would actually be easier/simpler to build on top of go native and if/when we feel like we need a different router to switch out to theirs. Rather than having to rip out another router we feel is too much overhead. I'll change it to a go-native implementation.